File: client/src/app/models/post.model.ts

export interface Post {
  id?: string,
  title: string
  permalink: string
  category: PostCategory
  postImgPath: string
  excerpt: string
  content: string
  isFeatured?: boolean
  views?: number
  status?: string,
    message?: string
  createdAt?: Date,
  updatedAt?: Date,
}

export interface PostCategory {
  catID: string
  catName: string,
}

export interface PhotoImage {
  id: string
  photo: string
  name: string
  size: string
  type: string
  photoPath: string,
  message: string,
}

export interface PostsData {
  featuredPosts: Post[];
  posts: Post[];
}

export interface PostsDetailsData {
  postDetails: Post;
  nextPosts: Post[];
}

File: Core/Models/PostModel.cs
using System.ComponentModel.DataAnnotations;

namespace Core.Models
{
  public class PostModel
  {
    [Key]
    public string? ID { get; set; }

    [Required]
    public string? Title { get; set; }

    [Required]
    public string? Permalink { get; set; }

    [Required]
    public PostCategoryModel? Category { get; set; }

    [Required]
    public string? PostImgPath { get; set; }

    [Required]
    public string? Excerpt { get; set; }

    [Required]
    public string? Content { get; set; }

    public Boolean IsFeatured { get; set; } = false;

    public int Views { get; set; } = 0;

    public string Status { get; set; } = "new";

    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public DateTime? UpdatedAt { get; set; }
  }
}
